feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/tindergov/basket/TinderGovBasketViewModel.kt

```diff
@@ -140,16 +143,23 @@ class TinderGovBasketViewModel(
             val availableToVoteReferenda = availableToVoteReferendaFlow.first()
 
             val removedReferenda = basket.filter { availableToVoteReferenda.contains(it.referendumId).not() }
+                .filter { interactor.basketItemValidForVote(it) }
+
             if (removedReferenda.isNotEmpty()) {
                 interactor.removeBasketItems(removedReferenda)
 
-                itemsWasRemovedFromBasketAction.awaitAction()
+                itemsWasRemovedFromBasketAction.awaitAction(getFormattedAmountAvailableToVote())
 
                 closeScreenIfBasketIsEmpty()
             }
```

Comment:
So we wont attempt to close the screen if there was some "invalid valid" referenda that was removed?